### DISCUSSION TITLE
OIDCCookieSameSite clarification

### DISCUSSION THREAD

**Discussion by OlivierBOEL** [OpenIDC/mod_auth_openidc#685] [Q&A]
Hello,

I'm puzzled about this [comment](https://github.com/zmartzone/mod_auth_openidc/blob/master/auth_openidc.conf#L502).
From my experice with mod_auth_openidc v2.3.7, it looks like SameSite attribute is set to strict after 10% of OIDCSessionInactivityTimeout  (with a max. of 60 seconds).
Could [this code](https://github.com/zmartzone/mod_auth_openidc/blob/c675949ca3609c9fe7dc0bccdaa093021ac59cbe/src/mod_auth_openidc.c#L1326)  explain what I've obeserved?
Or did I miss something?

Thanks,

Olivier

**Comment by zandbelt** [comment]
the idea is to set the cookie to `Lax` immediately after returning from the OP since that's the only value that works after a redirect; after that, i.e. upon the first update of the session, the cookie is set to `Strict`; the first update happens indeed faster than the inactivity timeout, but it is kind of cumbersome to explain all of that there since it is not that relevant...
